File: principlesJava/inherit.java
// Base class
public class Animal {
    
    private String name;

    
    public Animal(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

   
    public void makeSound() {
        System.out.println("Some generic animal sound");
    }
}

// Derived class
public class Dog extends Animal {
   
    public Dog(String name) {
        
        super(name);
    }

    
    @Override
    public void makeSound() {
        System.out.println("Bark");
    }


    public void fetch() {
        System.out.println(getName() + " is fetching the ball.");
    }
}

// Main class to demonstrate inheritance
public class Main {
    public static void main(String[] args) {
     
        Animal genericAnimal = new Animal("Generic Animal");
        
        genericAnimal.makeSound();

        
        Dog dog = new Dog("Buddy");
        
        dog.makeSound();
       
        dog.fetch();
    }
}

File: PrinciplesPython/Inherit.py
from django.db import models
#Inheritance allows a derived or child class to inherit attributes from and methods from another class called the superclass
#Base class
class Animal(models.Model):
    name = models.CharField(max_length=100)
    age = models.IntegerField()
    def make_sound(self):
        return "Some generic animal sound"

    def _str_(self):
        return f"{self.name}, {self.age}"
    

    # Derived class
class Dog(Animal):
    breed = models.CharField(max_length=100)
    def make_sound(self):
        return "Bark"

    def fetch(self):
        return f"{self.name} is fetching the ball."